### ISSUE TITLE
pskN modifier supports

### ISSUE THREAD

**Issue by cute** [rweather/noise-c#28]
protocol Noise_IKpsk2_25519_ChaChaPoly_BLAKE2s is invalid name.

**Comment by nazar-pc** [comment]
noise-c doesn't implement the latest version of the spec yet.
Hopefully, soon...

**Comment by rweather** [comment]
I apologise for how long this is taking.  My day job has been eating all my time over the last few months, leaving little time for noise-c.

I have pushed my in-progress rev32 work to the repository under the "rev32" branch.  I was in the middle of trying to get the cacophony test vectors for PSK's to work with noise-c.  There was also still some work to do on converting the way fallback works.

Any assistance to finish this off would be greatly appreciated.

**Comment by pallas** [comment]
FYI, I have a fix I'm working on for the above. The vectors pass now, I just need to clean some things up.

**Comment by pallas** [comment]
tl;dr is the following issue in the spec: when there are any PSKs, the E state goes from just calling MixHash to calling MixHash, then MixKey.
